lib/new_relic/agent/new_relic_service.rb

```diff
@@ -625,11 +626,10 @@ def decompress_response(response)
       # the ruby version and also zlib version if available since
       # that may cause corrupt compression if there is a problem.
       def user_agent
-        ruby_description = ''
-        # note the trailing space!
-        ruby_description << "(ruby #{::RUBY_VERSION} #{::RUBY_PLATFORM}) " if defined?(::RUBY_VERSION) && defined?(::RUBY_PLATFORM)
-        zlib_version = ''
-        zlib_version << "zlib/#{Zlib.zlib_version}" if defined?(::Zlib) && Zlib.respond_to?(:zlib_version)
+        if defined?(::RUBY_VERSION) && defined?(::RUBY_PLATFORM)
+          ruby_description = "(ruby #{::RUBY_VERSION} #{::RUBY_PLATFORM}) " # note the trailing space!
+        end
+        zlib_version = "zlib/#{Zlib.zlib_version}" if defined?(::Zlib) && Zlib.respond_to?(:zlib_version)
         "NewRelic-RubyAgent/#{NewRelic::VERSION::STRING} #{ruby_description}#{zlib_version}"
```

Comment:
If everything is eventually getting mushed into a single string, would it be plausible to remove the space from the end of `ruby_description` and add it on line 633 before `#{zlib_version}`?

Reply:
Well... the short answer is that we can't do it that way because the setting of `ruby_description` is guarded by a conditional. If we moved the trailing space to the big string on the last line of the method, we'd run the risk of having 2 spaces showing up together between the agent version number and the zlib version number when `ruby_description` is `nil`.

But the long answer is that I think we might want to reevaluate the conditional that allows `ruby_version` to potentially be `nil`. It seems like the check would always evaluate to `true` for our currently supported Rubies.